### PR TITLE
Enable usb tests on frdm_k64f

### PR DESCRIPTION
**Pull request by MaureenHelm**
- Enables usb tests on frdm_k64f
- Fixes a compile error in the kinetis usb device driver found when enabling these tests

### CODE REVIEW

File: drivers/usb/device/usb_dc_kinetis.c [hunk 1]
```diff
@@ -474,15 +474,15 @@ int usb_dc_ep_is_stalled(const u8_t ep, u8_t *const stalled)
...
-	if (SYS_LOG_LEVEL == SYS_LOG_LEVEL_INFO) {
-		u8_t bd_idx = get_bdt_idx(ep,
-				dev_data.ep_ctrl[ep_idx].status.in_odd);
-		bd_idx = bd_idx;
-		SYS_LOG_WRN("active bd ctrl: %x", bdt[bd_idx].set.bd_ctrl);
-		bd_idx = get_bdt_idx(ep,
-				~dev_data.ep_ctrl[ep_idx].status.in_odd);
-		SYS_LOG_WRN("next bd ctrl: %x", bdt[bd_idx].set.bd_ctrl);
-	}
+#if defined(CONFIG_SYS_LOG) && (SYS_LOG_LEVEL >= SYS_LOG_LEVEL_WARNING)
```
Comment (hakehuang) [inline drivers/usb/device/usb_dc_kinetis.c]: I think this change is due to the logic issue in include/logging/sys_log.h, meybe we can fix it there directly

File: tests/subsys/usb/bos/testcase.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
 tests:
   usb.bos:
-    platform_whitelist: native_posix
+    platform_whitelist: native_posix frdm_k64f
```
Comment (jfischer-phytec-iot) [inline tests/subsys/usb/bos/testcase.yaml]: I guess it should only be tested on native_posix, @finikorg ?

Comment (finikorg) [reply tests/subsys/usb/bos/testcase.yaml]: @jfischer-phytec-iot yes.

**Review by finikorg — CHANGES_REQUESTED** [review]
`bd_idx = bd_idx;` is here probably for masking some compiler warnings, right? We need to fix it.

File: drivers/usb/device/usb_dc_kinetis.c (hunk 1 above)
Comment (MaureenHelm) [reply drivers/usb/device/usb_dc_kinetis.c]: What is the logic issue in `include/logging/sys_log.h`?

File: tests/subsys/usb/bos/testcase.yaml (hunk 2 above)
Comment (MaureenHelm) [reply tests/subsys/usb/bos/testcase.yaml]: Why?

File: drivers/usb/device/usb_dc_kinetis.c (hunk 1 above)
Comment (hakehuang) [reply drivers/usb/device/usb_dc_kinetis.c]: the SYS_LOG_WRN is defined in sys_log.h, but it wil lbe none defined in below condition
define CONIG_SYS_LOG  and SYS_LOG_LEVEL < SYS_LOG_LEVEL_WARNING.
so in the definition of SYS_LOG_WRN, it shall not be 
{code}
#if (SYS_LOG_LEVEL >= SYS_LOG_LEVEL_WARNING)
#define SYS_LOG_WRN(...) LOG_COLOR(SYS_LOG_TAG_WRN,                     \
        SYS_LOG_COLOR_YELLOW, ##__VA_ARGS__)
#endif
{code}
but
{code}
#if (SYS_LOG_LEVEL >= SYS_LOG_LEVEL_WARNING)
#define SYS_LOG_WRN(...) LOG_COLOR(SYS_LOG_TAG_WRN,                     \
        SYS_LOG_COLOR_YELLOW, ##__VA_ARGS__)
#else  
 #define SYS_LOG_WRN(...) { ; }
#endif

{code}

File: tests/subsys/usb/bos/testcase.yaml (hunk 2 above)
Comment (finikorg) [reply tests/subsys/usb/bos/testcase.yaml]: Those (bos and osdesc) are unit tests, they are not "build tests" which only need to be built but run tests, they need to be executed on the host.

Comment (MaureenHelm) [reply tests/subsys/usb/bos/testcase.yaml]: With `--device-testing`, they **are** executed on the host. In this case, the host is frdm_k64f.

Comment (jfischer-phytec-iot) [reply tests/subsys/usb/bos/testcase.yaml]: that's good, I'm for it

**Review by jfischer-no — APPROVED** [review]

File: drivers/usb/device/usb_dc_kinetis.c [hunk 3]
```diff
@@ -474,15 +474,15 @@ int usb_dc_ep_is_stalled(const u8_t ep, u8_t *const stalled)
...
+	u8_t bd_idx = get_bdt_idx(ep,
+			dev_data.ep_ctrl[ep_idx].status.in_odd);
+	bd_idx = bd_idx;
```
[the remark this answers is not in the document]
Comment (MaureenHelm) [reply drivers/usb/device/usb_dc_kinetis.c]: I wondered that too, but didn't want to remove it without knowing why it's there. @jfischer-phytec-iot ?

[the remark this answers is not in the document]
Comment (jfischer-phytec-iot) [reply drivers/usb/device/usb_dc_kinetis.c]: I do not know it anymore, please remove it.

[the remark this answers is not in the document]
Comment (MaureenHelm) [reply drivers/usb/device/usb_dc_kinetis.c]: Done

**Review by finikorg — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**